File: scripts/generate-icons.js
import { writeFileSync, mkdirSync } from 'fs';
import { createCanvas } from 'canvas';

const sizes = [16, 32, 72, 96, 128, 144, 152, 192, 384, 512];
const dir = 'public/icons';
mkdirSync(dir, { recursive: true });

for (const size of sizes) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Background
  const gradient = ctx.createLinearGradient(0, 0, size, size);
  gradient.addColorStop(0, '#572580');
  gradient.addColorStop(1, '#8553ac');
  ctx.fillStyle = gradient;
  ctx.fillRect(0, 0, size, size);
  
  // Warning triangle (white)
  const cx = size / 2;
  const cy = size / 2;
  const triSize = size * 0.45;
  const h = triSize * Math.sqrt(3) / 2;
  
  ctx.beginPath();
  ctx.moveTo(cx, cy - h * 2/3);
  ctx.lineTo(cx + triSize/2, cy + h/3);
  ctx.lineTo(cx - triSize/2, cy + h/3);
  ctx.closePath();
  ctx.fillStyle = 'white';
  ctx.fill();
  
  // Exclamation mark
  const markSize = size * 0.15;
  ctx.fillStyle = '#572580';
  ctx.beginPath();
  ctx.roundRect(cx - markSize/2, cy - h/3, markSize, h * 0.6, markSize/2);
  ctx.fill();
  
  ctx.beginPath();
  ctx.arc(cx, cy + h/3 - markSize * 1.2, markSize/2, 0, Math.PI * 2);
  ctx.fill();
  
  const buffer = canvas.toBuffer('image/png');
  writeFileSync(`${dir}/${size}x${size}.png`, buffer);
  console.log(`Generated ${size}x${size}.png`);
}

// Also create 128x128@2x.png
const canvas2x = createCanvas(256, 256);
const ctx2x = canvas2x.getContext('2d');
const gradient2x = ctx2x.createLinearGradient(0, 0, 256, 256);
gradient2x.addColorStop(0, '#572580');
gradient2x.addColorStop(1, '#8553ac');
ctx2x.fillStyle = gradient2x;
ctx2x.fillRect(0, 0, 256, 256);

const cx2 = 128;
const cy2 = 128;
const triSize2 = 115;
const h2 = triSize2 * Math.sqrt(3) / 2;

ctx2x.beginPath();
ctx2x.moveTo(cx2, cy2 - h2 * 2/3);
ctx2x.lineTo(cx2 + triSize2/2, cy2 + h2/3);
ctx2x.lineTo(cx2 - triSize2/2, cy2 + h2/3);
ctx2x.closePath();
ctx2x.fillStyle = 'white';
ctx2x.fill();

const markSize2 = 38;
ctx2x.fillStyle = '#572580';
ctx2x.beginPath();
ctx2x.roundRect(cx2 - markSize2/2, cy2 - h2/3, markSize2, h2 * 0.6, markSize2/2);
ctx2x.fill();
ctx2x.beginPath();
ctx2x.arc(cx2, cy2 + h2/3 - markSize2 * 1.2, markSize2/2, 0, Math.PI * 2);
ctx2x.fill();

writeFileSync(`${dir}/128x128@2x.png`, canvas2x.toBuffer('image/png'));
console.log('Generated 128x128@2x.png');

// Create ICO and ICNS using the 256px version
const icoCanvas = createCanvas(256, 256);
const icoCtx = icoCanvas.getContext('2d');
icoCtx.drawImage(canvas2x, 0, 0);
writeFileSync('src-tauri/icons/icon.ico', icoCanvas.toBuffer('image/png'));
console.log('Generated icon.ico (as png, will need conversion)');

const icnsCanvas = createCanvas(1024, 1024);
const icnsCtx = icnsCanvas.getContext('2d');
icnsCtx.drawImage(canvas2x, 0, 0, 1024, 1024);
writeFileSync('src-tauri/icons/icon.icns', icnsCanvas.toBuffer('image/png'));
console.log('Generated icon.icns (as png, will need conversion)');

console.log('All icons generated. Note: .ico and .icns need proper conversion tools.');

File: scripts/generate-icons.py
"""Generate app icons using PIL (no native deps)."""
from PIL import Image, ImageDraw
import os

sizes = [16, 32, 72, 96, 128, 144, 152, 192, 384, 512]
os.makedirs('public/icons', exist_ok=True)
os.makedirs('src-tauri/icons', exist_ok=True)

for size in sizes:
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Background gradient (approximated as solid for simplicity)
    bg_color = (87, 37, 128)  # #572580
    draw.rectangle([0, 0, size, size], fill=bg_color)
    
    # Warning triangle
    cx = size / 2
    cy = size / 2
    tri_size = size * 0.45
    h = tri_size * (3**0.5) / 2
    
    triangle = [
        (cx, cy - h * 2/3),
        (cx + tri_size/2, cy + h/3),
        (cx - tri_size/2, cy + h/3)
    ]
    draw.polygon(triangle, fill='white')
    
    # Exclamation mark
    mark_size = size * 0.15
    # Bar
    bar_left = cx - mark_size/2
    bar_top = cy - h/3
    bar_right = cx + mark_size/2
    bar_bottom = cy - h/3 + h * 0.6
    draw.rounded_rectangle([bar_left, bar_top, bar_right, bar_bottom], radius=mark_size/2, fill='#572580')
    
    # Dot
    dot_cy = cy + h/3 - mark_size * 1.2
    draw.ellipse([cx - mark_size/2, dot_cy - mark_size/2, cx + mark_size/2, dot_cy + mark_size/2], fill='#572580')
    
    img.save(f'public/icons/{size}x{size}.png')
    print(f'Generated public/icons/{size}x{size}.png')

# 128x128@2x (256x256)
img2x = Image.new('RGBA', (256, 256), (0, 0, 0, 0))
draw2x = ImageDraw.Draw(img2x)
draw2x.rectangle([0, 0, 256, 256], fill=(87, 37, 128))
cx2, cy2 = 128, 128
tri2 = 115
h2 = tri2 * (3**0.5) / 2
triangle2 = [(cx2, cy2 - h2*2/3), (cx2+tri2/2, cy2+h2/3), (cx2-tri2/2, cy2+h2/3)]
draw2x.polygon(triangle2, fill='white')
mark2 = 38
draw2x.rounded_rectangle([cx2-mark2/2, cy2-h2/3, cx2+mark2/2, cy2-h2/3+h2*0.6], radius=mark2/2, fill='#572580')
dot_cy2 = cy2 + h2/3 - mark2 * 1.2
draw2x.ellipse([cx2-mark2/2, dot_cy2-mark2/2, cx2+mark2/2, dot_cy2+mark2/2], fill='#572580')
img2x.save('public/icons/128x128@2x.png')
print('Generated public/icons/128x128@2x.png')

# For Tauri - create multiple sizes in ICO format
ico_sizes = [(16,16), (32,32), (48,48), (64,64), (128,128), (256,256)]
ico_images = []
for w, h in ico_sizes:
    ico_img = Image.new('RGBA', (w, h), (0, 0, 0, 0))
    draw = ImageDraw.Draw(ico_img)
    draw.rectangle([0, 0, w, h], fill=(87, 37, 128))
    cx, cy = w/2, h/2
    tri = min(w, h) * 0.45
    h_tri = tri * (3**0.5) / 2
    triangle = [(cx, cy - h_tri*2/3), (cx+tri/2, cy+h_tri/3), (cx-tri/2, cy+h_tri/3)]
    draw.polygon(triangle, fill='white')
    mark = min(w, h) * 0.15
    draw.rounded_rectangle([cx-mark/2, cy-h_tri/3, cx+mark/2, cy-h_tri/3+h_tri*0.6], radius=mark/2, fill='#572580')
    dot_cy = cy + h_tri/3 - mark * 1.2
    draw.ellipse([cx-mark/2, dot_cy-mark/2, cx+mark/2, dot_cy+mark/2], fill='#572580')
    ico_images.append(ico_img)

ico_images[0].save('src-tauri/icons/icon.ico', format='ICO', sizes=[(img.width, img.height) for img in ico_images])
print('Generated src-tauri/icons/icon.ico')

# ICNS - save as 1024x1024 PNG (macOS can use this directly or we convert)
icns_img = Image.new('RGBA', (1024, 1024), (0, 0, 0, 0))
draw = ImageDraw.Draw(icns_img)
draw.rectangle([0, 0, 1024, 1024], fill=(87, 37, 128))
cx, cy = 512, 512
tri = 460
h_tri = tri * (3**0.5) / 2
triangle = [(cx, cy - h_tri*2/3), (cx+tri/2, cy+h_tri/3), (cx-tri/2, cy+h_tri/3)]
draw.polygon(triangle, fill='white')
mark = 150
draw.rounded_rectangle([cx-mark/2, cy-h_tri/3, cx+mark/2, cy-h_tri/3+h_tri*0.6], radius=mark/2, fill='#572580')
dot_cy = cy + h_tri/3 - mark * 1.2
draw.ellipse([cx-mark/2, dot_cy-mark/2, cx+mark/2, dot_cy+mark/2], fill='#572580')
icns_img.save('src-tauri/icons/icon.png')
print('Generated src-tauri/icons/icon.png (1024x1024 for ICNS conversion)')

print('\\nAll icons generated.')
print('Note: For proper .icns, use `iconutil -c icns iconset` on macOS with an iconset folder.')
print('The .ico file is properly formatted.')